File: backend_safeher/app/schemas/incident.py
from datetime import datetime
from pydantic import BaseModel


class IncidentCreate(BaseModel):
    latitude: float
    longitude: float
    type: str = "other"
    severity: str = "medium"
    description: str | None = None


class IncidentOut(BaseModel):
    id: str
    latitude: float
    longitude: float
    type: str
    severity: str
    description: str | None
    reported_at: datetime

    class Config:
        from_attributes = True


class SafetyStatusOut(BaseModel):
    status: str  # safe | caution | high_risk
    score: int
    reason: str
    incident_count: int
    incidents: list[IncidentOut]

File: safehere_flutterapp/lib/models/incident.dart
class Incident {
  final String id;
  final double latitude;
  final double longitude;
  final String type;
  final String severity; // low | medium | high
  final String? description;
  final String reportedAt;

  Incident({
    required this.id,
    required this.latitude,
    required this.longitude,
    required this.type,
    required this.severity,
    this.description,
    required this.reportedAt,
  });

  factory Incident.fromJson(Map<String, dynamic> json) => Incident(
        id: json['id'],
        latitude: (json['latitude'] as num).toDouble(),
        longitude: (json['longitude'] as num).toDouble(),
        type: json['type'],
        severity: json['severity'],
        description: json['description'],
        reportedAt: json['reported_at'],
      );
}

class SafetyStatus {
  final String status; // safe | caution | high_risk
  final int score;
  final String reason;
  final int incidentCount;
  final List<Incident> incidents;

  SafetyStatus({
    required this.status,
    required this.score,
    required this.reason,
    required this.incidentCount,
    required this.incidents,
  });

  factory SafetyStatus.fromJson(Map<String, dynamic> json) => SafetyStatus(
        status: json['status'],
        score: json['score'],
        reason: json['reason'],
        incidentCount: json['incident_count'] ?? 0,
        incidents: (json['incidents'] as List? ?? []).map((e) => Incident.fromJson(e)).toList(),
      );
}
